### PR TITLE
AcceptOfferController Logging/issue 102

### PR DESCRIPTION
**Pull request by anish97IND**

### CODE REVIEW

**Comment by magwas** [comment]
You have conflicts. If you need help, call your mentor.

**Comment by magwas** [comment]
You have a mutation coverage to fix. call me if you need help.

**Comment by szirbucz** [comment]
Please resolve your conflicts

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    Mockito.verify(loggerService)
+        .info(
+            Mockito.eq(AcceptOfferControllerTestData.SERVICE_CALL_NAME), captorString.capture()
+        );
+    assertEquals(
+        AcceptOfferController.class.getName(),
+        captorString.getValue()
+    );
+  }
+
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java:47]: You can earn much more with this. the success log is missing.

File: src/main/java/com/kodekonveyor/work_request/WorkRequestConstants.java [hunk 2]
```diff
@@ -58,5 +58,5 @@
   public static final String OWNER_ID_REGEX = "[0-9]+";
   public static final int MIN_PRICE = 1;
 
-  public static final String WORK_REQUEST_RECEIVED = "workrequest.received";
```
Comment (magwas) [inline src/main/java/com/kodekonveyor/work_request/WorkRequestConstants.java:61]: I may have mistakenly deleted this line when resolving the conflict for you. If the build fails, put it back.

**Comment by magwas** [comment]
updated, pull before push

File: src/main/java/com/kodekonveyor/work_request/offer/AcceptOfferController.java [hunk 3]
```diff
@@ -22,16 +24,41 @@
...
   @PutMapping("/accept/{offerId}")
   public WorkRequestDTO call(final long offerId) {
 
+    loggerService.info(
+        WorkRequestConstants.SERVICE_CALL_NAME, this.getClass().getName()
+    );
+    loggerService.info(WorkRequestConstants.FIND_OFFER, offerId);
     final OfferEntity offerEntity =
         offerEntityRepository.findById(offerId).get();
+    loggerService.debug(
```
Comment (magwas) [inline src/main/java/com/kodekonveyor/work_request/offer/AcceptOfferController.java:39]: you don't have to log each single step

File: src/main/java/com/kodekonveyor/work_request/offer/AcceptOfferController.java [hunk 4]
```diff
@@ -22,16 +24,41 @@
...
+        WorkRequestConstants.FIND_OFFER_STATUS, offerEntity.getId(),
+        WorkRequestConstants.SUCCESS
+    );
     final WorkRequestEntity workRequest = offerEntity.getWorkRequest();
+    loggerService.info(
```
Comment (magwas) [inline src/main/java/com/kodekonveyor/work_request/offer/AcceptOfferController.java:44]: you don't have to log each single step

File: src/main/java/com/kodekonveyor/work_request/offer/AcceptOfferController.java [hunk 5]
```diff
@@ -22,16 +24,41 @@
...
+        WorkRequestConstants.SET_WORK_REQUEST_ENUM, WorkRequestStatusEnum.AGREED
+    );
     workRequest.setStatus(WorkRequestStatusEnum.AGREED);
+    loggerService.debug(
```
Comment (magwas) [inline src/main/java/com/kodekonveyor/work_request/offer/AcceptOfferController.java:48]: you don't have to log each single step

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  @Test
+  @DisplayName(
+    "The Find work request log"
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java:52]: please use a full English sentence understandable by end user.

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        );
+    assertEquals(
+        OfferDTOTestData.get().getId(),
+        captorString.getValue()
+    );
+  }
+
+  @Test
+  @DisplayName(
+    "Save work request log"
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java:67]: please use a full English sentence understandable by end user.

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+        );
+    assertEquals(
+        WorkRequestEntityTestData.getStatusAgreed(),
+        captorString.getValue()
+    );
+  }
+
+  @Test
+  @DisplayName(
+    "return work request DTO success"
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java:85]: please use a full English sentence understandable by end user.

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+        );
+    assertEquals(
+        WorkRequestDTOTestData.getStatusAgreedAndProvider(),
+        captorString.getValue()
+    );
+  }
+
+  @Test
+  @DisplayName(
+    "return work request log"
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java:105]: please use a full English sentence understandable by end user.

File: src/main/java/com/kodekonveyor/work_request/offer/AcceptOfferController.java [hunk 10]
```diff
@@ -22,16 +24,32 @@
...
+        WorkRequestConstants.SERVICE_CALL_NAME, this.getClass().getName()
+    );
+    loggerService.info(WorkRequestConstants.FIND_OFFER, offerId);
     final OfferEntity offerEntity =
         offerEntityRepository.findById(offerId).get();
+
     final WorkRequestEntity workRequest = offerEntity.getWorkRequest();
+
     workRequest.setStatus(WorkRequestStatusEnum.AGREED);
+    loggerService.info(WorkRequestConstants.SAVE_WORK_REQUEST, workRequest);
```
Comment (magwas) [inline src/main/java/com/kodekonveyor/work_request/offer/AcceptOfferController.java:43]: not needed

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+        );
+    assertEquals(
+        AcceptOfferController.class.getName(),
+        captorString.getValue()
+    );
+  }
+
+  @Test
+  @DisplayName(
+    "The Find work request for a particula offer id is logged"
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java:52]: "The id of the offer is logged"

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+        );
+    assertEquals(
+        WorkRequestEntityTestData.getStatusAgreed(),
+        captorString.getValue()
+    );
+  }
+
+  @Test
+  @DisplayName(
+    "After successful completion of all the process, work request DTO is successfully returned and is logged"
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerLoggingTest.java:85]: return is not verified here, (and should not be verified for this behaviour)

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerTestBase.java [hunk 13]
```diff
@@ -21,9 +22,16 @@
   @Mock
   WorkRequestRepository workRequestRepository;
   @Mock
+  Logger loggerService;
+  @Mock
   AuthenticatedUserService authenticatedUserService;
+
   @Captor
   ArgumentCaptor<WorkRequestEntity> captorEntity;
+  @Captor
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerTestBase.java:31]: the captors should be in the test.
(if it raises a PMD violation, then probably we should rewrite the pmd rule)

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerTestData.java [hunk 14]
```diff
@@ -0,0 +1,18 @@
+package com.kodekonveyor.work_request.offer;
+
+public class AcceptOfferControllerTestData {
+
+  public static final String SERVICE_CALL_NAME = "Service Call Name: {}";
+  public static final String FIND_OFFER = "Find offer by ID: {}";
+  public static final String FIND_OFFER_STATUS = "Find offer by ID: {} {}";
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerTestData.java:7]: It should be FIND_OFFER+" {}"

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerTestData.java [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+  public static final String SUCCESS = "SUCCESS";
+  public static final String SET_WORK_REQUEST_ENUM =
+      "Set work request status {}";
+  public static final String SET_WORK_REQUEST_ENUM_STATUS =
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerTestData.java:11]: same as above

File: src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerTestData.java [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+      "Set work request status {} {}";
+  public static final String SAVE_WORK_REQUEST = "Save work request: {}";
+  public static final String RETURN_WORK_REQUEST_DTO =
+      "Returning work request DTO: {}";
+  public static final String RETURN_WORK_REQUEST_DTO_STATUS =
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/AcceptOfferControllerTestData.java:16]: same as above

**Comment by magwas** [comment]
closing due to inactivity. please regrab if you want to work on it.

**Pull request closed without merging**